File: Core/src/main/java/com/hqq/core/ui/binding/BaseBindingActivity.java
package com.hqq.core.ui.binding;

import android.view.View;
import android.view.ViewGroup;

import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;

import com.hqq.core.ui.base.BaseActivity;
import com.hqq.core.ui.builder.ICreateRootView;

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.core.ui.binding
 * @FileName :   BaseBindingActivity
 * @Date : 2020/7/30 0030  上午 10:41
 * @Email : qiqiang213@gmail.com
 * @Descrive :
 */
public abstract class BaseBindingActivity<T extends ViewDataBinding> extends BaseActivity implements ICreateRootView.IBanding {
    protected T mBinding;

    /**
     * 禁止 子类继承使用 保证走的都是getLayoutView方法
     *
     * @return
     */
    @Override
    public final int getLayoutViewId() {
        return 0;
    }

    @Override
    public View getLayoutView(ViewGroup parent) {
        mBinding = DataBindingUtil.inflate(getLayoutInflater(), getLayoutId(), parent, false);
        mBinding.setLifecycleOwner(this);

        return mBinding.getRoot();
    }

}

File: Core/src/main/java/com/hqq/core/ui/binding/BaseBindingActivity.kt
package com.hqq.core.ui.binding

import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import com.hqq.core.ui.base.BaseActivity
import com.hqq.core.ui.base.ICreateRootView.IBanding

/**
 * @Author : huangqiqiang
 * @Package : com.hqq.core.ui.binding
 * @FileName :   BaseBindingActivity
 * @Date : 2020/7/30 0030  上午 10:41
 * @Email : qiqiang213@gmail.com
 * @Descrive :
 */
abstract class BaseBindingActivity<T : ViewDataBinding>
    : BaseActivity(), IBanding {

    lateinit var binding: T

    /**
     * 禁止 子类继承使用 保证走的都是getLayoutView方法
     * @return
     */
    final override val layoutViewId: Int = 0
    /**
     * 绑定Binding
     */
    override fun getLayoutView(parent: ViewGroup): View {
        binding = DataBindingUtil.inflate<T>(layoutInflater, layoutId, parent, false)
        binding.lifecycleOwner = this
        return binding.root
    }
}